src/java/org/apache/cassandra/net/OutboundConnectionInitiator.java

```diff
@@ -211,6 +204,15 @@ public void initChannel(SocketChannel channel) throws Exception
                 logger.trace("creating outbound netty SslContext: context={}, engine={}", sslContext.getClass().getName(), sslHandler.engine().getClass().getName());
                 pipeline.addFirst("ssl", sslHandler);
             }
+            else
+            {
+                // If a connection is SSL based we don't need to make additional authentication when making a client
+                // connection. Trusted roots in the client's truststore would suffice for authentication.
+                //
+                // Outbound connection is a client connection we don't need to extract identity from server certificate and authenticate.
+                // Add this ServerAuthenticationHandler only when encryption is not enabled just for backward compatibility.
+                pipeline.addLast("server-authentication", new ServerAuthenticationHandler());
```

Comment:
I think you should call the authenticator for outbound too. This changes the behavior of always authenticating on outbound regardless of SSL.

It may not be true for all operators that the trusted root check is sufficient and they may also want to check the outbound IP addresses too, the age of the certificate or log the authentication attempt. Adding the direction to the `authenticate` method would make it easy to implement different behaviors on inbound/outbound.